### PR TITLE
feat: add job listing

### PR DESCRIPTION
**Pull request by Arnthorny**
## Description
Create an API endpoint to allow admins post new job listings.
This endpoint will validate the job posting and then add it to the database.
​

The changes include:
 
* **Endpoint Addition:** Added `POST /api/v1/jobs` for posting new jobs.
* **Error Handling:** Implemented checks for invalid data and unauthorized access.
* **Testing:** Added `pytest` cases to ensure functionality and edge cases are covered.
 
 
## Related Issue (Link to issue ticket)
[Issue Ticket](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/490)

 ## Motivation and Context
This feature is essential to allow admins post new jobs on the site.
 
 ## How Has This Been Tested?
 * **Unit Tests:** Added tests to validate the addition of jobs to the database, authorization failures and invalid request body.
* **Test Environment:** Tests were run using a mocked database instance to avoid interference with the production environment.

Tests include:
 
* Adding a job to the database successfully.
* Handling invalid request body
* Handling unauthorized access.

### Successful Request
image

### Invalid request body
image

## Types of changes
* [x]  New feature (non-breaking change which adds functionality)
 
 ## Checklist:
 * [x]  My code follows the code style of this project.
 * [x]  My change requires a change to the documentation.
 * [x]  I have updated the documentation accordingly.
 * [x]  I have read the **CONTRIBUTING** document.
 * [x]  I have added tests to cover my changes.
 * [x]  All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/jobs.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        new_job = job_service.create(db, job_full)
+        logger.info(f"Job Listing created successfully {new_job.id}")
+
+        return success_response(
+            message = "Job listing created successfully",
+            status_code = 201,
+            data = jsonable_encoder(JobCreateResponseSchema.model_validate(new_job))
+        )
+
+    except Exception as e:
```
Comment (joboy-dev) [inline api/v1/routes/jobs.py:56]: Exceptions are already being handled by the app. No need for this

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes and I will merge

**Comment by Arnthorny** [comment]
> Make changes and I will merge

@joboy-dev Changes implemented

**Comment by joboy-dev** [comment]
Merge conflict!!

**Comment by Arnthorny** [comment]
> Merge conflict!!

Merge conflict resolved @joboy-dev

**Comment by Arnthorny** [comment]
> Merge conflict!!

Conflict resolved.

Following the new conventions, this endpoint should only be accessible by an admin.
@joboy-dev

**Review by Goketech — APPROVED** [review]

**Pull request merged**